### ISSUE TITLE
[Pytorch] Generation instead of Translation 

### ISSUE THREAD

**Issue by SoyGema** [SoyGema/The-Lord-of-The-Words-The-two-frameworks#6]
#### Context 
Fine-tuning **english-hindi** Translation model with [t5-small](https://huggingface.co/SoyGema/t5-small) and [opus100](https://huggingface.co/datasets/opus100) dataset
Running the example run_translation.py from transformers repository.
Small modification for making the dataset a little bit smaller for testing end-to-end

Full Script here. 

Checked recommendations from [README.md](https://github.com/huggingface/transformers/blob/0afa5071bd84e44301750fdc594e33db102cf374/examples/pytorch/translation/README.md?plain=1#L81) when using T5 family models 

- [X] 1. Add --source_prefix flag 
- [X] 2. Change 3 flags accordingly `--source_lang` , `--target_lang` and `--source_prefix`

```
python run_translation.py \
    --model_name_or_path t5-small \
    --do_train \
    --do_eval \
    --source_lang en \
    --target_lang hi \
    --source_prefix "translate English to Hindi: " \
    --dataset_name opus100 \
    --dataset_config_name en-hi \
    --output_dir=/tmp/english-hindi \
    --per_device_train_batch_size=4 \
    --per_device_eval_batch_size=4 \
    --overwrite_output_dir \
    --num_train_epochs=3 \
    --push_to_hub=True \
    --predict_with_generate=True
    --report_to_all
    --do_predict
```
Trace of model card . Once the model is trained
```
[INFO|modelcard.py:452] 2023-09-02 23:08:32,386 >> Dropping the following result as it does not have all the necessary fields:
{'task': {'name': 'Sequence-to-sequence Language Modeling', 'type': 'text2text-generation'}, 'dataset': {'name': 'opus100', 'type': 'opus100', 'config': 'en-hi', 'split': 'validation', 'args': 'en-hi'}}
```

Somehow the task is uploaded in Hub as  **text-generation** and not as a **translation task**.
Inference shows text-generation as well, And the model card seems to point at that too.
During search , I visited / Read [forum](https://discuss.huggingface.co/t/trainer-vs-seq2seqtrainer/3145), but I think it makes reference to the BLEU generation metric , and not the task.
<img width="1308" alt="text-generation-instead-translation" src="https://github.com/SoyGema/The-Lord-of-The-Words-The-two-frameworks/assets/24204714/abdee474-3629-4f86-b669-f7b2f2e6209a">

Any ideas of how I shall set the translation task correctly ?

**Comment by SoyGema** [comment]
Read [forum](https://discuss.huggingface.co/t/trainer-vs-seq2seqtrainer/3145)
